core/templates/domain/statistics/state-top-answers-stats-object.factory.ts

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Domain object for state top answers stats.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import { AnswerStatsObjectFactory, AnswerStats, AnswerStatsBackendDict } from
+  'domain/exploration/AnswerStatsObjectFactory';
+
+interface StateInteractionIds {
+  [state: string]: string;
```

Comment:
Should be stateName not state, right? Ditto elsewhere.

Be careful to distinguish between these two concepts.

Reply:
done